presto-main/src/main/java/com/facebook/presto/sql/gen/PageFunctionCompiler.java

```diff
@@ -319,18 +338,22 @@ private static MethodDefinition generateProcessMethod(
         return method;
     }
 
-    private static BytecodeBlock generateCheckYieldBlock(Variable thisVariable, Variable index, FieldDefinition yieldSignal, FieldDefinition nextIndexOrPosition)
+    private BytecodeBlock generateCheckYieldBlock(Variable thisVariable, Variable start, Variable index, FieldDefinition yieldSignal, FieldDefinition nextIndexOrPosition)
     {
         return new BytecodeBlock()
-                .comment("if (yieldSignal.isSet())")
+                .comment("if (System.nanoTime() - start > yieldCheckPeriodNanos)")
```

Comment:
Maybe I'm missing it, but where is start updated?

Reply:
`start` is initialized on L309.